src/pages/index.js

```diff
@@ -1,120 +1,267 @@
+  profAboutSelector: selectors.profAbout,
+  profAvatarSelector: selectors.profAvatar,
+});
+
+const editProfInfo = new PopupWithForm(
+  selectors.editProfFormModal,
+  handleProfEditSubmit,
+  "Save",
+  "Saving..."
 );
-/* --------------------- Edit Card -------------------- */
 
-// Create the edit form instance
-const editFormPopup = new PopupWithForm(selectors.editFormPopup, (values) => {
-  // Add the form's input to the profile section
-  userInfo.setUserInfo(values.name, values.description);
-});
+function handleProfEditSubmit({ name, about }) {
+  editProfInfo.showLoading();
+  api
+    .setProfileInfoAPI(name, about)
+    .then((dataAPI) => {
+      userInfo.setProfileInfo({
+        name: name,
+        about: about,
+      });
+
+      editProfInfo.close();
+    })
+    .finally(() => {
+      editProfInfo.hideLoading();
+    });
+}
 
-// Open the modal when users click on the edit button
-profileEditButton.addEventListener("click", () => {
-  // Get profile info and add to the form fields
-  const profileInfo = userInfo.getUserInfo();
+editProfButton.addEventListener("click", () => {
+  // Open the avatar edit modal form
+  editProfInfo.open();
 
-  // Add the profile info on the page to the form's fields
-  profileTitleInput.value = profileInfo.name;
-  profileDescriptionInput.value = profileInfo.description;
+  // Add the profile info from the page to the form's fields
+  const profInfo = userInfo.getProfileInfo();
+  profNameInput.value = profInfo.name;
+  profAboutInput.value = profInfo.about;
 
-  // Open modal
-  editFormPopup.open();
+  // Reset validation for the avatar edit modal form
   editFormValidator.resetForm();
 });
 
-// Set edit form event listeners
-editFormPopup.setEventListeners();
+editProfInfo.setEventListeners();
 
-/* ------------------- Add Card ------------------- */
+/* -------------------Create Initial Cards --------------------- */
+
+// Receives card data(cardData) from API and renders data to cards.
+api.getInfoAPI().then(([cardData, userData]) => {
+  cardSection = new Section(
+    {
+      items: cardData,
+      renderer: (cardData) => {
+        // Renders a new card
+        const cardList = createCard(cardData, userData._id);
 
-const addFormPopup = new PopupWithForm(selectors.addFormPopup, (formData) => {
-  // Create a new card
-  const newCard = createCard(formData);
+        // Filters delete button access based on matching user ID.
+        const userID = userData._id;
+        const cardOwnerID = cardData.owner._id;
 
-  // Add the new card to the section
-  cardSection.addItem(newCard);
+        if (userID !== cardOwnerID) {
+          card._removeDeleteButton();
+        }
+        // Adds each rendered card to the DOM.
+        cardSection.addItem(cardList);
+      },
+    },
+    selectors.cardsList
+  );
+
+  // Renders each card via Section.js to the DOM.
+  cardSection.renderItems();
+  const name = userData.name;
+  const about = userData.about;
```

Comment:
You don't need that because `userData` is a ready object


```js
  cardSection.renderItems();

  userInfo.setProfileInfo(userData);
  userInfo.setAvatar(userData.avatar);

```